The version pin is irrelevant to how we build and deliver zope.interface.
https://github.com/zopefoundation/zope.interface/issues/333

--- zope.interface-7.1.1/pyproject.toml
+++ zope.interface-7.1.1/pyproject.toml
@@ -3,7 +3,7 @@
 # https://github.com/zopefoundation/meta/tree/master/config/c-code
 
 [build-system]
-requires = ["setuptools<74"]
+requires = ["setuptools"]
 build-backend = "setuptools.build_meta"
 
 [tool.coverage.run]
